### ISSUE TITLE
Table of Contents: Add IDs to h4-h6 headings to allow direct linking

### ISSUE THREAD

**Issue by StevenDufresne** [WordPress/wporg-mu-plugins#625]
We link headings based on their appearance in the Table of Contents ([source](https://github.com/WordPress/wporg-mu-plugins/blob/5c6616a1b2093dcf1e51e097a4330d5622395d36/mu-plugins/blocks/table-of-contents/index.php#L95)).

However, we want to link all headings for easy sharing.

**How to reproduce**
1. Visit a [developer resource page](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-scripts/#advanced-usage)
2. Notice not all headings have shareable link 
  a. Try looking for "Default webpack config" in the link about

**Comment by ryelle** [comment]
+1 for making the headings linkable, but by default this would also add those to the Table of Contents, and I don't think that would be helpful.

**Comment by StevenDufresne** [comment]
I agree, being linked in the document shouldn't make them show up in the Table of Contents.
